ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -161,6 +183,56 @@ showEBBs AnalysisEnv { db, registry } = do
           _otherwise ->
             return () -- Skip regular blocks
 
+{-------------------------------------------------------------------------------
+  Analysis: store a ledger at specific slot
+-------------------------------------------------------------------------------}
+
+storeLedgerAtSlot ::
+     forall blk .
+     ( LgrDbSerialiseConstraints blk
+     , HasAnalysis blk
+     , LedgerSupportsProtocol blk
+     )
+  => SlotNo -> Analysis blk
+storeLedgerAtSlot slotNo (AnalysisEnv { db, registry, initLedger, cfg, hasFS }) = do
+    putStrLn $ "About to store snapshot of a ledger at " <>
+               show slotNo <> " " <>
+               "this will take a while..."
+    void $ processAll db registry GetBlock initLedger process
+  where
+    process :: ExtLedgerState blk -> blk -> IO (ExtLedgerState blk)
+    process oldLedger blk = do
+      let ledgerCfg     = ExtLedgerCfg cfg
+          appliedResult = tickThenApplyLedgerResult ledgerCfg blk oldLedger
+          newLedger     = either (error . show) lrResult $ runExcept $ appliedResult
+      if slotNo == blockSlot blk then
```

Comment:
what about:

```haskell
when (slotNo == ..) $ storeNewLedger
return newLedger
```

however the current logic depends on whether we should stop sooner.

Reply:
oh, this is much better